utils/notificationhubServicePartition.ts

```diff
@@ -32,18 +32,18 @@ export const getNHLegacyConfig = (
  * @returns
  */
 export const testShaForPartitionRegex = (
-  regex: string,
+  regex: RegExp | string,
   sha: InstallationId
-): boolean => new RegExp(regex).test(sha);
+): boolean => (typeof regex === "string" ? new RegExp(regex) : regex).test(sha);
 
 /**
  * @returns A connection string based on NH namespace and sharedAccessKey provided
  */
 export const buildNHConnectionString = ({
   namespace,
   sharedAccessKey
-}: NotificationHubPartitions): string =>
-  `Endpoint=sb://${namespace}.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=${sharedAccessKey}`;
+}: NotificationHubPartition): NonEmptyString =>
+  `Endpoint=sb://${namespace}.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=${sharedAccessKey}` as NonEmptyString;
```

Comment:
It makes sense to force-cast to `NonEmptyString` here, as we are reasonably sure we're not passing an empty string. 
Conversely, it is not safe to make such an assumption from a caller perspective.